### ISSUE TITLE
[v9.10.0] `@intlify/message-compiler` publishing missing TypeScript reference

### ISSUE THREAD

**Issue by shengslogar** [intlify/vue-i18n-next#1760]
### Reporting a bug?

Beginning in v9.10.0 and persistent in v9.10.1, `@intlify/message-compiler/dist/message-compiler.d.ts` is publishing the following TS reference:

```
/// <reference types="node_modules/.pnpm/source-map-js@1.0.2/node_modules/source-map-js/source-map" />
```

https://www.npmjs.com/package/@intlify/message-compiler/v/9.10.0?activeTab=code

This causes TS to fail out when `compilerOptions.skipLibCheck` is set to `false` in your `tsconfig.json`.

### Expected behavior

I scanned a diff of [v9.9.1...v9.10.0](https://github.com/intlify/vue-i18n-next/compare/v9.9.1...v9.10.0) and couldn't see any obvious reason why this line was added, but my naive opinion is we shouldn't be assuming `.pnpm` is being used.

_**EDIT:** I'm seeing now that PNPM is clearly the development tool for this repo. Not sure what the best practice is here for publishing references._

Expected behavior is to not have any TS references to files outside of the installed package and its dependencies.

Workaround for now is setting `compilerOptions.skipLibCheck` to `true`.

### Reproduction

https://stackblitz.com/edit/vitejs-vite-1dmm1b?file=package.json

`npm run build` is automatically run, and you will see the following terminal output:

```
node_modules/@intlify/message-compiler/dist/message-compiler.d.ts:1:23 - error TS2688: Cannot find type definition file for 'node_modules/.pnpm/source-map-js@1.0.2/node_modules/source-map-js/source-map'.

1 /// <reference types="node_modules/.pnpm/source-map-js@1.0.2/node_modules/source-map-js/source-map" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Found 1 error in node_modules/@intlify/message-compiler/dist/message-compiler.d.ts:1
```

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.3 - /usr/local/bin/pnpm
  npmPackages:
    @vitejs/plugin-vue: ^4.2.3 => 4.6.2 
    vite: ^4.4.9 => 4.5.2 
    vue: ^3.3.4 => 3.4.21 
    vue-i18n: 9.10.0 => 9.10.0
    vue-tsc: ^1.8.8 => 1.8.27
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n-next/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussions](https://github.com/intlify/vue-i18n-next/discussions)

**Comment by xterler159** [comment]
I am getting the same behaviour with this package. @shengslogar Your workaround is working for now, thanks

**Comment by kazupon** [comment]
Hmm 🤔, Apparently, after bumping to TS 5.3, the reference type is now embedded in the build file.
I am using api-extractor from d.ts. Maybe it is an issue it has.

**Comment by shengslogar** [comment]
@kazupon This was [patched in v9.10.2](https://stackblitz.com/edit/vitejs-vite-dgp1ps?file=package.json) but [regressed in v9.11.0](https://stackblitz.com/edit/vitejs-vite-5sfnez?file=package.json).

Updated error output:

```
node_modules/@intlify/message-compiler/dist/message-compiler.d.ts:1:23 - error TS2688: Cannot find type definition file for '.pnpm/source-map-js@1.2.0/node_modules/source-map-js'.

1 /// <reference types=".pnpm/source-map-js@1.2.0/node_modules/source-map-js" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@intlify/message-compiler/dist/message-compiler.d.ts:2:23 - error TS2688: Cannot find type definition file for 'node_modules/.pnpm/source-map-js@1.0.2/node_modules/source-map-js/source-map'.

2 /// <reference types="node_modules/.pnpm/source-map-js@1.0.2/node_modules/source-map-js/source-map" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: node_modules/@intlify/message-compiler/dist/message-compiler.d.ts:1
```

**Comment by kazupon** [comment]
Thanks!
I'm going to fix regression issue as new issue.
